main.go

```diff
@@ -223,6 +223,11 @@ func handlePost(res http.ResponseWriter, req *http.Request) {
 
 	date, _ := time.ParseInLocation("2006-01-02", dateStr, loc)
 
+	if date.Weekday() != 4 && !time.IsZero() {
```

Comment:
we cannot neglect the `date.Weekday()`-check, because there might be non-zero dates that are not thursdays (e.g. due to typo in the date). One could do two ifs, though, but I'd prefer the one-if-version for readability.